liblangutil/Scanner.h

```diff
@@ -62,15 +62,12 @@
 namespace solidity::langutil
 {
 
-class AstRawString;
-class AstValueFactory;
-class ParserRecorder;
-
 enum class ScannerKind
 {
 	Solidity,
 	Yul,
-	ExperimentalSolidity
+	ExperimentalSolidity,
+	SourceLocationComment
```

Comment:
```suggestion
	SpecialComment,
```

Maybe this would be a better name? I think we'll in the future reuse this mode when reimplementing regexes with scanner, so a more general name would make more sense. Most of those cases are generally in comments and inside quoted strings I think (anything that's proper syntax gets parsed in other modes).